### PR TITLE
CI: pkg/management checks: gofmt/go vet/golint

### PR DESCRIPTION
**Pull request by ahmetalpbalkan**
Yet another stab at #112 to have `go vet`/`golint` as part of the CI. Nearly 280 fixes in the `management` package.

**My biggest concern is** around some structs where we changed field name (e.g. DnsServers→DNSServers) that might impact the XML marshaling/unmarshaling in the request/response. Please pay extra attn to those...

Currently there are **112 `golint` warnings** remaining, I **suppressed** those.

101 of them are regarding missing godocs on exported types, e.g.:

    exported type DNSServer should have comment or be unexported

and 11 of them are regarding stuttering package name in the types, e.g.:

    type name will be used as storageservice.StorageServiceClient by other packages, and that stutters; consider calling this Client
    type name will be used as virtualnetwork.VirtualNetworkConfiguration by other packages, and that stutters; consider calling this Configuration

Other than that this PR includes some comment (godoc) tweaks.

@paulmey please review sometime.

Signed-off-by: Ahmet Alp Balkan

### CODE REVIEW

**Comment by svanharmelen** [comment]
@ahmetalpbalkan nice!

As for the DnsServer->DNSServer like changes, that should not matter as log as the struct tags are inline with what the API expects. Tags, if available, are used instead of the actual field names while doing XML marshaling/unmarshaling. So this should be just fine...

**Comment by ahmetalpbalkan** [comment]
Yeah I know but I appreciate if some extra eyes can take a look because I might have missed some of those.

**Comment by paulmey** [comment]
Overall, LGTM. after you rebase I can take an extra look at the XML tags...?

**Comment by ahmetalpbalkan** [comment]
Rebased and pushed.

File: management/publishSettings.go [hunk 1]
```diff
@@ -82,15 +82,15 @@ type publishProfile struct {
 	XMLName               xml.Name       `xml:"PublishProfile"`
 	SchemaVersion         string         `xml:",attr"`
 	PublishMethod         string         `xml:",attr"`
-	Url                   string         `xml:",attr"`
+	RL                    string         `xml:"Url,attr"`
```
Comment (paulmey) [inline management/publishSettings.go]: RL->URL?

File: management/virtualnetwork/entities.go [hunk 2]
```diff
@@ -6,61 +6,61 @@ import (
 	"github.com/Azure/azure-sdk-for-go/management"
 )
 
-const xmlNamespace = "http://schemas.microsoft.com/ServiceHosting/2011/07/NetworkConfiguration"
-const xmlNamespaceXsd = "http://www.w3.org/2001/XMLSchema"
-const xmlNamespaceXsi = "http://www.w3.org/2001/XMLSchema-instance"
+const xmlNamespace = "http:// schemas.microsoft.com/ServiceHosting/2011/07/NetworkConfiguration"
+const xmlNamespaceXsd = "http:// www.w3.org/2001/XMLSchema"
+const xmlNamespaceXsi = "http:// www.w3.org/2001/XMLSchema-instance"
```
Comment (paulmey) [inline management/virtualnetwork/entities.go]: These three lines seem find/replace errors "//"->"// " ?

File: management/virtualnetwork/entities.go [hunk 3]
```diff
@@ -6,61 +6,61 @@ import (
...
 type VirtualNetworkClient struct {
 	client management.Client
 }
 
-//NetworkConfiguration represents the network configuration for an entire Azure
-//subscription. TODO: Nicer builder methods for these that abstract away the
-//underlying structure
+// NetworkConfiguration represents the network configuration for an entire Azure
+// subscription. TODO: Nicer builder methods for these that abstract away the
+// underlying structure
```
Comment (paulmey) [inline management/virtualnetwork/entities.go]: If you agree with the TODO, let's make it into an issue...

Comment (ahmetalpbalkan) [inline management/virtualnetwork/entities.go]: I didn't add it it was there. :) Feel free to.

**Comment by paulmey** [comment]
LGTM

**Pull request merged**